services/web/server/src/simcore_service_webserver/login/handlers.py

```diff
@@ -379,7 +539,12 @@ async def email_confirmation(request: web.Request):
             await db.update_user(user, {"status": ACTIVE})
             await db.delete_confirmation(confirmation)
             log.debug("User %s registered", user)
-            redirect_url = redirect_url.with_fragment("?registered=true")
+            if settings.LOGIN_2FA_REQUIRED:
+                redirect_url = redirect_url.with_fragment(
+                    f"2fa-verify?email={user['email']}"
+                )
+            else:
+                redirect_url = redirect_url.with_fragment("?registered=true")
```

Comment:
This is not needed any more, this is checked in the login method

Reply:
but it is tmp necessary until we change the front-end, right?

Reply:
No, the frontend knows (and will actually won't) nothing about the ```2fa-verify``` page

Reply:
Ahh, i thought you were referred to ``"?registered=true"``